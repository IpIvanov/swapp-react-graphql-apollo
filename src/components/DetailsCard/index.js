import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import { Card, CardMedia, CardContent, Typography, makeStyles } from '@material-ui/core';
import styles from './styles';
import ThemeContext from '../../contexts/ThemeContext';

const DetailsCard = ({ title, image, details }) => {
  const {
    theme,
  } = useContext(ThemeContext);
  const useStyles = makeStyles(styles);
  const classes = useStyles({ theme });

  return (
    <Card className={classes.card}>
      <Typography gutterBottom variant="h5" component="h2" className={classes.title}>
        {title}
      </Typography>
      <CardMedia
        className={classes.cover}
        image={image}
        title={title}
      />
      <CardContent className={classes.content}>
        {details.map((detail) => (
          <Typography
            variant="body2"
            color="textSecondary"
            component="p"
            className={classes.description}
            align="left"
            key={detail.key}
          >
            {`${detail.key}: `}
            <span className={classes.detailValue}>
              {detail.value}
              {' '}
              {detail.key === 'Cost' && 'credits'}
            </span>
          </Typography>
        ))}
      </CardContent>
    </Card>
  );
};

DetailsCard.propTypes = {
  title: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  details: PropTypes.array.isRequired,
};

export default DetailsCard;
