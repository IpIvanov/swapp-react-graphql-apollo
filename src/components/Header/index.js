import React from 'react';
import { useHistory, withRouter, NavLink } from 'react-router-dom';
import { Input } from '@material-ui/icons';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core';
import styles from './styles';

const Header = () => {
  const useStyles = makeStyles(styles);
  const classes = useStyles();
  const history = useHistory();

  return (
    <div className={`${classes.container}`}>
      <Typography
        className={classes.linkHome}
      >
        SWAPP
      </Typography>
      <div className={classes.rightHeader}>
        <NavLink
          to="/episodes"
          className={classes.link}
          underline="none"
        >
          Episodes
        </NavLink>
        <NavLink
          to="/characters"
          className={classes.link}
          underline="none"
        >
          Characters
        </NavLink>
        <div className={classes.exit}>
          <Input
            onClick={() => { history.push('/'); }}
          />
        </div>
      </div>
    </div>
  );
};
export default withRouter(Header);
