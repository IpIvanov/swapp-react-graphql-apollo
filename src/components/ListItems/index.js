import React from 'react';
import { Link } from 'react-router-dom';
import { makeStyles, Typography, Card, Grid, Box, Button, CardMedia, CardContent } from '@material-ui/core';
import styles from './styles';

const ListItems = ({ listItems, loadMoreIsVisible, loadMoreHandler, linkTo, mdColumns }) => {
  const useStyles = makeStyles(styles);
  const classes = useStyles();

  return (
    <Grid container display="flex" direction="row">
      <Grid container spacing={2}>
        {listItems.map(({ name, id, image }, i) => (
          <Grid item xs={12} md={mdColumns} key={i.toString()}>
            <Link to={`/${linkTo}/${id}`} className={classes.linkTo}>
              <Card className={classes.card}>
                <CardMedia
                  className={classes.cover}
                  image={image}
                  title={name}
                  src={name}
                />
                <CardContent className={classes.content}>
                  <Typography component="h5" variant="h5" className={classes.name}>
                    {name}
                  </Typography>
                </CardContent>
              </Card>
            </Link>
          </Grid>
        ))}
      </Grid>
      {loadMoreIsVisible && (
        <Box display="flex" justifyContent="center" alignItems="center" flex="1">
          <Button
            color="primary"
            variant="outlined"
            size="small"
            className={classes.button}
            onClick={loadMoreHandler}
          >
            Load More
          </Button>
        </Box>
      )}
    </Grid>
  );
};

export default ListItems;
