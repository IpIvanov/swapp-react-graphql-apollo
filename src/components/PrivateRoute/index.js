/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { Redirect, Route } from 'react-router-dom';

const PrivateRoute = ({ component: Component, publicRoute, isAuthorized, ...rest }) => (
  <Route
    {...rest}
    render={(props) => (
      isAuthorized ? (<Component {...props} />)
        : (<Redirect to={{ pathname: publicRoute }} />)
    )}
  />
);

export default PrivateRoute;
