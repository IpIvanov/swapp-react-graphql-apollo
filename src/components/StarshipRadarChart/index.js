/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import PropTypes from 'prop-types';
import { RadarChart, CircularGridLines } from 'react-vis';

const StarshipRadarChart = (
  { starshipData: { starshipClass, maxAtmosphericSpeed, cost, crew, hyperdriveRating, maxMLPerHour },
    allStarshipsData },
) => {
  const data = allStarshipsData.filter((starship) => starship.node.starshipClass === starshipClass)
    .map((
      { node: { cost: Cost, crew: Crew, hyperdriveRating: hdRating, maxMLPerHour: maxMlPHour, maxAtmosphericSpeed: maSpeed } },
    ) => ({ Cost, Crew, 'HyperD Rat.': hdRating, 'Max ML/h': maxMlPHour, 'Max Atm. Speed': maSpeed }));
  let DOMAIN = [
    { name: 'Max Atm. Speed', domain: [0, Math.max(...data.map((o) => o['Max Atm. Speed']), 0)], tickFormat: (t) => t },
    { name: 'Cost', domain: [0, Math.max(...data.map((o) => o.Cost), 0)] },
    { name: 'Crew', domain: [0, Math.max(...data.map((o) => o.Crew), 0)] },
    { name: 'HyperD Rat.', domain: [0, Math.max(...data.map((o) => o['HyperD Rat.']), 0) + 100] },
    { name: 'Max ML/h', domain: [0, Math.max(...data.map((o) => o['Max ML/h']), 0)] },
  ];

  const dataObject = () => {
    const obj = {};
    if (maxAtmosphericSpeed) {
      obj['Max Atm. Speed'] = maxAtmosphericSpeed;
    }
    if (cost) {
      obj.Cost = cost;
    }
    if (crew) {
      obj.Crew = crew;
    }
    if (hyperdriveRating) {
      obj['HyperD Rat.'] = hyperdriveRating;
    }
    if (maxMLPerHour) {
      obj['Max ML/h'] = maxMLPerHour;
    }

    return obj;
  };

  const DATA = [
    dataObject(),
  ];

  DOMAIN = DOMAIN.map((item) => Object.keys(DATA[0]).map((title) => {
    if (title === item.name) {
      return item;
    }

    return undefined;
  })).flat().filter((item) => item !== undefined);

  return (
    <RadarChart
      animation
      data={DATA}
      domains={DOMAIN}
      style={{
        polygons: {
          fillOpacity: 0.4,
          strokeWidth: 2,
          strokeOpacity: 1,
        },
        axes: {
          text: {
            opacity: 1,
          },
          line: {
            strokeWidth: 2,
          },
        },
        labels: {
          textAnchor: 'middle',
          fontSize: 12,
          fontWeight: 600,
          fill: '#fff',
        },
      }}
      margin={{
        left: 50,
        top: 40,
        right: 50,
      }}
      width={400}
      height={400}
    >
      <CircularGridLines
        style={{ stroke: '#333' }}
        tickValues={[...new Array(10)].map((v, i) => i / 10 - 1)}
      />
    </RadarChart>
  );
};

StarshipRadarChart.propTypes = {
  starshipData: PropTypes.object.isRequired,
  allStarshipsData: PropTypes.array.isRequired,
};

export default StarshipRadarChart;
