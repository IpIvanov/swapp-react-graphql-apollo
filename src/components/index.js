import Header from './Header';
import PrivateRoute from './PrivateRoute';
import ListItems from './ListItems';
import DetailsCard from './DetailsCard';

export { Header, PrivateRoute, ListItems, DetailsCard };
