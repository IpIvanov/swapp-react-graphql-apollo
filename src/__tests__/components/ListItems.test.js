/* eslint global-require: 0 */
import React from 'react';
import { shallow } from 'enzyme';
import { ListItems } from '../../components';

jest.mock('react-router-dom', () => ({
  useHistory: () => ({
    push: jest.fn(),
  }),
}));

describe('Components | ListItems', () => {
  it('renders without crashing', () => {
    const listItemsMock = [{
      id: 'people.1',
      image: 'image-link.jpeg',
      name: 'Luke Skywalker',
    }, {
      id: 'people.1',
      image: 'image-link.jpeg',
      name: 'Luke Skywalker',
    }];
    const wrapper = shallow(<ListItems
      listItems={listItemsMock}
      loadMoreIsVisible
      loadMoreHandler={() => { }}
      linkToUrl="/some-route"
      mdColumns={4}
    />);

    expect(wrapper).toMatchSnapshot();
  });
});
