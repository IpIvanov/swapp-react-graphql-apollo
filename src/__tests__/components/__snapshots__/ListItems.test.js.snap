// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components | ListItems renders without crashing 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
  direction="row"
  display="flex"
>
  <WithStyles(ForwardRef(Grid))
    container={true}
    spacing={2}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      key="0"
      md={4}
      xs={12}
    >
      <WithStyles(ForwardRef(Link))
        className="makeStyles-linkTo-6"
        href="/"
        onClick={[Function]}
        underline="none"
      >
        <WithStyles(ForwardRef(Card))
          className="makeStyles-card-1"
        >
          <WithStyles(ForwardRef(CardMedia))
            className="makeStyles-cover-3"
            image="image-link.jpeg"
            src="Luke Skywalker"
            title="Luke Skywalker"
          />
          <WithStyles(ForwardRef(CardContent))
            className="makeStyles-content-2"
          >
            <WithStyles(ForwardRef(Typography))
              className="makeStyles-name-4"
              component="h5"
              variant="h5"
            >
              Luke Skywalker
            </WithStyles(ForwardRef(Typography))>
          </WithStyles(ForwardRef(CardContent))>
        </WithStyles(ForwardRef(Card))>
      </WithStyles(ForwardRef(Link))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      key="1"
      md={4}
      xs={12}
    >
      <WithStyles(ForwardRef(Link))
        className="makeStyles-linkTo-6"
        href="/"
        onClick={[Function]}
        underline="none"
      >
        <WithStyles(ForwardRef(Card))
          className="makeStyles-card-1"
        >
          <WithStyles(ForwardRef(CardMedia))
            className="makeStyles-cover-3"
            image="image-link.jpeg"
            src="Luke Skywalker"
            title="Luke Skywalker"
          />
          <WithStyles(ForwardRef(CardContent))
            className="makeStyles-content-2"
          >
            <WithStyles(ForwardRef(Typography))
              className="makeStyles-name-4"
              component="h5"
              variant="h5"
            >
              Luke Skywalker
            </WithStyles(ForwardRef(Typography))>
          </WithStyles(ForwardRef(CardContent))>
        </WithStyles(ForwardRef(Card))>
      </WithStyles(ForwardRef(Link))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
  <Styled(MuiBox)
    alignItems="center"
    display="flex"
    flex="1"
    justifyContent="center"
  >
    <WithStyles(ForwardRef(Button))
      className="makeStyles-button-5"
      color="primary"
      onClick={[Function]}
      size="small"
      variant="outlined"
    >
      Load More
    </WithStyles(ForwardRef(Button))>
  </Styled(MuiBox)>
</WithStyles(ForwardRef(Grid))>
`;
